src/gui/AstroCalcDialog.cpp

```diff
@@ -3225,24 +3230,19 @@ localSEparameter localSolarEclipse(double JD,int contact,bool central) {
 	double lat = static_cast<double>(core->getCurrentLocation().latitude);
 	double lon = static_cast<double>(core->getCurrentLocation().longitude);
 	double elevation = static_cast<double>(core->getCurrentLocation().altitude);
-	lat = lat * M_PI_180;
 	lon = -(lon);
 	double L = 0.;
 
+	static SolarSystem* ssystem = GETSTELMODULE(SolarSystem);
+	Vec4d geocentricCoords = ssystem->getEarth()->getRectangularCoordinates(lon,lat,elevation);
+	static const double earthRadius = ssystem->getEarth()->getEquatorialRadius();
+	static const double rc = geocentricCoords[0]/earthRadius; // rhoCosPhiPrime
```

Comment:
Uh, `rc `and `rs `must not be static! When you change location, the `geocentricCoords` should also change! (static here means "set at first run and keep results until program exits")

Reply:
Oops. fixed. It should be okay now.